boa/src/builtins/array/mod.rs

```diff
@@ -414,25 +414,25 @@ impl Array {
         if len == 0 {
             this.set_field("length", 0);
             // Since length is 0, this will be an Undefined value
-            return Ok(this.get_field(0.to_string()));
+            return Ok(this.get_field(0));
```

Comment:
What is the difference between this and just returning `Ok(Value::undefined())`? I think the later would more-closely represent the spec, and it would be easier to understand.

Reply:
Yes. checking the spec this is not what the spec does I'll fix it :) , this PR was not focused on making the `Array` methods spec compliant, If I put to much in a PR (which already I did looking at the title :sweat_smile: ) it makes it harder to maintain and for others to review.

Reply:
Definitely xD I just noticed it while skimming through the diff and looked weird.